react/features/base/react/components/web/Watermarks.js

```diff
@@ -173,18 +191,37 @@ class Watermarks extends Component<Props, State> {
      */
     _renderJitsiWatermark() {
         let reactElement = null;
-
-        if (this.state.showJitsiWatermark
-                || (this.props._isGuest
-                    && this.state.showJitsiWatermarkForGuests)) {
-            reactElement = <div className = 'watermark leftwatermark' />;
-
-            const { jitsiWatermarkLink } = this.state;
-
-            if (jitsiWatermarkLink) {
+        const {
+            showJitsiWatermark,
+            showJitsiWatermarkForGuests,
+            jitsiWatermarkLink
+        } = this.state;
+        const {
+            _isGuest,
+            _customLogoUrl,
+            _customLogoLink,
+            _readyToDisplayJitsiWatermark
+        } = this.props;
+
+        if (_readyToDisplayJitsiWatermark && (showJitsiWatermark
```

Comment:
we can simplify all these checks by computing a single `showJitsiWatermark` in `mapStateToProps`

Reply:
I have moved this in a separate method.